src/dune_file_watcher/dune_file_watcher.ml

```diff
@@ -0,0 +1,241 @@
+open! Stdune
+
+type t =
+  { pid : Pid.t
+  ; wait_for_watches_established : unit -> unit
+  }
+
+let pid t = t.pid
+
+let buffer_capacity = 65536
+
+(* Fixed-size buffer for reading line-by-line from file descriptors. Bug:
+   deadlocks if there's a line longer than the capacity of the buffer. TODO: use
+   Out_channel? *)
```

Comment:
```suggestion
   In_channel? *)
```

Reply:
Thanks. Fixed.

Reply:
Using `in_channel` seems fine. IIRC, we tried that initially and had some issues, maybe related to blocking. But the code was a bit different, in particular we didn't have the debouncing thread. So maybe we can make it work. In any case, 64k should be fine on most systems given the max path length.